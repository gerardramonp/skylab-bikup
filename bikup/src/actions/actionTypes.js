export default {
    LOAD_BIKES: 'LOAD_BIKES',
    LOAD_BIKE_BY_ID: 'LOAD_BIKE_BY_ID'
};
