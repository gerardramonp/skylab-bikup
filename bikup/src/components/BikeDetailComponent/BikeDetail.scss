$light-bg-color: #eeeeee;
$light-highlight-color: #d7d7d7;
$main-color: #32e0c4;

.bike-detail {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 24px 24px;
}

h2 {
    margin-bottom: 24px;
}

.bike-detail__upper {
    display: flex;
    flex-direction: row;
    width: 100%;
    justify-content: space-between;
    margin-bottom: 24px;
}

.bike-detail__km-hours {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 24px;
    width: 100%;
}

.km-hours {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 0 5%;
    width: 130px;
    padding: 12px;
    background-color: $light-highlight-color;
    border-radius: 5px;
}

.km-hours__title {
    font-weight: bold;
    margin-bottom: 5px;
}

.bike-detail__components {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
}

.bike-detail__details {
    display: flex;
    flex-direction: row;
    justify-content: center;
    width: 100%;
}

.card-decoration {
    height: inherit;
    width: 2%;
    border-top-left-radius: 5px;
    border-bottom-left-radius: 5px;
    background-color: $main-color;
}

.details__content {
    display: flex;
    flex-direction: row;
    align-items: center;
    width: 88%;
    background-color: $light-highlight-color;
    border-top-right-radius: 5px;
    border-bottom-right-radius: 5px;
    padding: 12px;
}

.details__type-img {
    height: 50px;
    width: 50px;
    margin-right: 12px;
}

.details__info {
    display: flex;
    flex-direction: column;
}

.details__info p {
    padding: 5px 0px;
}

.info__type {
    font-weight: bold;
}
