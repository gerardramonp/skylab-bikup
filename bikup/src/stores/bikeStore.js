import dispatcher from '../dispatcher';
import actionTypes from '../actions/actionTypes';
import { EventEmitter } from 'events';

const CHANGE_EVENT = 'change';

let _bike = {};

class BikeStore extends EventEmitter {
	addChangeListener(callback) {
		this.on(CHANGE_EVENT, callback);
	}

	removeChangeListener(callback) {
		this.removeListener(CHANGE_EVENT, callback);
	}

	emitChange() {
		this.emit(CHANGE_EVENT);
	}

	setBikeDetail(bikeDetail) {
		_bike = bikeDetail;
	}

	getBikeDetail() {
		return _bike;
	}
}

const bikeStore = new BikeStore();

dispatcher.register((action) => {
	switch (action.type) {
		case actionTypes.LOAD_BIKE_BY_ID:
			bikeStore.setBikeDetail(action.data);
			bikeStore.emitChange();
			break;
		default:
			console.log(`There is no action with type: ${action.type}`);
			break;
	}
});

export default bikeStore;
