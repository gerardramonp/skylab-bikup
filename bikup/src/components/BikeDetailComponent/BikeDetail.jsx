import React, { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import bikeStore from '../../stores/bikeStore';
import { loadBikeById } from '../../actions/bikeActions';

import CompoCard from './CompoCardComponent/CompoCard';
import StandardAside from '../StandardAside/StandardAside';

import '../../App.scss';
import './BikeDetail.scss';

function BikeDetail(props) {
    const [bikeDetail, setBikeDetail] = useState({});

    useEffect(() => {
        bikeStore.addChangeListener(onChange);
        if (!bikeDetail.bikeId) {
            setBikeDetail(bikeStore.getBikeDetail());
        }
        return () => bikeStore.removeChangeListener(onChange);
    });

    function onChange() {
        setBikeDetail(bikeStore.getBikeDetail());
    }

    return (
        bikeDetail && (
            <div className="general-container">
                <div className="desktop">
                    <StandardAside />
                </div>

                <div className="bike-detail">
                    <div className="bike-detail__upper mobile">
                        <NavLink to="/bikes">Back</NavLink>
                        <p className="upper__edit">Edit</p>
                    </div>

                    <div className="bike-detail__head">
                        <h2 className="head__bikename">
                            {bikeDetail.bikeName || 'Your Bike'}
                        </h2>
                        <div className="separator"></div>

                        <div className="bike-detail__km-hours">
                            <div className="km-hours km-hours__km">
                                <p className="km-hours__title">Total KM</p>
                                <div className="separator-small"></div>
                                <p className="km__value">
                                    {bikeDetail.bikeTotalMeters / 1000}
                                </p>
                            </div>
                            <div className="km-hours__hours km-hours">
                                <p className="km-hours__title">Total Hours</p>
                                <div className="separator-small"></div>
                                <p className="km-hours__value">
                                    {Math.floor(
                                        bikeDetail.bikeTotalMinutes / 60
                                    )}
                                </p>
                            </div>
                            <button className="float__button">+ Workout</button>
                        </div>
                    </div>

                    <div className="bike-detail__components">
                        {bikeDetail.components &&
                            bikeDetail.components.map((compo) => (
                                <CompoCard key={compo.name} compoInfo={compo} />
                            ))}
                    </div>

                    <div className="bike-detail__details">
                        <div className="card-decoration"></div>
                        <div className="details__content">
                            <img
                                src="https://image.flaticon.com/icons/svg/829/829906.svg"
                                alt="bike type"
                                className="details__type-img"
                            />
                            <div className="details__info">
                                <p className="info__type">
                                    {bikeDetail.bikeType}
                                </p>
                                <p>
                                    {bikeDetail.bikeBrand || 'Unknown Brand'} -{' '}
                                    {bikeDetail.bikeModel || 'Unknown Model'}
                                </p>
                                <p>
                                    Driving Style: {bikeDetail.bikeDriveStyle}
                                </p>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    );
}

export default BikeDetail;
