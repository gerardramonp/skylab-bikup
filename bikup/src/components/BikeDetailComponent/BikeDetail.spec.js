import React from 'react';
import renderer from 'react-test-renderer';
import BikeDetail from './BikeDetail';
import { BrowserRouter as Router } from 'react-router-dom';
import dispatcher from '../../dispatcher';
import bikeStore from '../../stores/bikeStore';
import actionTypes from '../../actions/actionTypes';

describe('Bike Detail', () => {
	xit('should match', () => {
		let tree = renderer.create(
			<Router>
				<BikeDetail />
			</Router>
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	xit('should load bike list', () => {
		const bikeTestName = 'testName';
		dispatcher.dispatch({
			type: actionTypes.LOAD_BIKE_BY_ID,
			data: { bikeName: bikeTestName },
		});

		const bike = bikeStore.getBikeDetail();

		expect(bike.bikeName).toEqual(bikeTestName);
	});
});
