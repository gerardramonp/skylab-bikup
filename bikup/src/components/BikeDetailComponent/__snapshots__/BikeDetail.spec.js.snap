// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bike Detail Should call onChange function 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="header__mobile"
    >
      <img
        alt="logo"
        className="logo"
        src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
      />
      <div
        className="mobile__nav"
      >
        <a
          aria-current={null}
          href="/search"
          onClick={[Function]}
        >
          <img
            alt="ExploreMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/214746d15b751953f02eb252fb879874/magnifying-glass.png"
          />
        </a>
        <a
          aria-current={null}
          href="/bikes"
          onClick={[Function]}
        >
          <img
            alt="MyBikesMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ca1db79d00b304cb995a7491f0ce71a0/bike.png"
          />
        </a>
        <a
          aria-current={null}
          href="/user"
          onClick={[Function]}
        >
          <img
            alt="ProfileMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/e60b6c14d78fa03c5a1ae02b95f9cbca/user.png"
          />
        </a>
      </div>
      <img
        alt="menu"
        className="nav__menu"
        src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ea9ccf0f7a5fe6c8d066bb89f204f504/open-menu.png"
      />
    </div>
    <div
      className="header__desktop"
    >
      <img
        alt="logo"
        className="logo"
        src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
      />
      <p
        className="appName"
      >
        bikUP
      </p>
      <ul
        className="header__navigation"
      >
        <a
          aria-current={null}
          className="navigation__item"
          href="/bikes"
          onClick={[Function]}
        >
          My Bikes
        </a>
        <a
          aria-current={null}
          className="navigation__item"
          href="/search"
          onClick={[Function]}
        >
          Explore
        </a>
        <a
          aria-current={null}
          className="navigation__item"
          href="/user"
          onClick={[Function]}
        >
          Profile
        </a>
      </ul>
      <div
        className="flex-spacer"
      />
      <button
        className="login__button login__button--header"
      >
        Log In
      </button>
    </div>
  </header>,
  <div
    className="general-container"
  >
    <div
      className="desktop"
    >
      <div
        className="standardaside"
      >
        <div
          className="standardaside__card-container"
        >
          <h3
            className="title"
          >
            Challenges
          </h3>
          <div
            className="standardaside__card"
          >
            <div>
              <div
                className="challenges__item"
              >
                <div
                  className="item__title"
                >
                  <img
                    alt="trophy"
                    className="title__img"
                    src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
                  />
                  - 
                  BIrlfriend
                   -
                </div>
                <p>
                  Travel 5000km with a bike
                </p>
              </div>
              <div
                className="separator"
              />
            </div>
            <div>
              <div
                className="challenges__item"
              >
                <div
                  className="item__title"
                >
                  <img
                    alt="trophy"
                    className="title__img"
                    src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
                  />
                  - 
                  CompoManiac
                   -
                </div>
                <p>
                  Change the same component 3 times
                </p>
              </div>
              <div
                className="separator"
              />
            </div>
          </div>
        </div>
        <div
          className="standardaside__card-container"
        >
          <h3
            className="title"
          >
            Give Away
          </h3>
          <div
            className="standardaside__card"
          >
            <p>
              something free
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="bike-detail"
    >
      <div
        className="bike-detail__upper mobile"
      >
        <a
          aria-current={null}
          href="/bikes"
          onClick={[Function]}
        >
          Back
        </a>
        <p
          className="upper__edit"
        >
          Edit
        </p>
      </div>
      <div
        className="bike-detail__head"
      >
        <h2
          className="head__bikename"
        >
          Your Bike
        </h2>
        <div
          className="separator"
        />
        <div
          className="bike-detail__km-hours"
        >
          <div
            className="km-hours km-hours__km"
          >
            <p
              className="km-hours__title"
            >
              Total KM
            </p>
            <div
              className="separator-small"
            />
            <p
              className="km__value"
            >
              NaN
            </p>
          </div>
          <div
            className="km-hours__hours km-hours"
          >
            <p
              className="km-hours__title"
            >
              Total Hours
            </p>
            <div
              className="separator-small"
            />
            <p
              className="km-hours__value"
            >
              NaN
            </p>
          </div>
          <button
            className="float__button"
          >
            + Workout
          </button>
        </div>
      </div>
      <div
        className="bike-detail__components"
      />
      <div
        className="bike-detail__details"
      >
        <div
          className="details__content"
        >
          <img
            alt="bike type"
            className="details__type-img"
            src="https://image.flaticon.com/icons/svg/829/829906.svg"
          />
          <div
            className="details__info"
          >
            <p
              className="info__type"
            />
            <p>
              Unknown Brand
               
              -
               
              Unknown Model
            </p>
            <p>
              Driving Style:
               
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Bike Detail should match 1`] = `
Array [
  <div
    className="mobile__menu"
  >
    <div
      className="menu__decorator"
      onClick={[Function]}
    >
      <div
        className="decorator__line"
      />
    </div>
    <ul
      className="menu__list"
    >
      <li
        className="menu__option"
        onClick={[Function]}
      >
        <img
          alt="Logout"
          className="option__img"
          src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f5a496d520b7b130a5d4819/7f0d35433cdacef755bec9d18cbcce2c/logout.png"
        />
        <p
          className="option__p"
        >
          Logout
        </p>
      </li>
    </ul>
  </div>,
  <header
    className="header"
  >
    <div
      className="header__mobile"
    >
      <div
        className="mobile__nav"
      >
        <img
          alt="logo"
          className="logo"
          src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
        />
        <a
          aria-current={null}
          href="/bikes"
          onClick={[Function]}
        >
          <img
            alt="ExploreMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/214746d15b751953f02eb252fb879874/magnifying-glass.png"
          />
        </a>
        <a
          aria-current={null}
          href="/bikes"
          onClick={[Function]}
        >
          <img
            alt="MyBikesMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ca1db79d00b304cb995a7491f0ce71a0/bike.png"
          />
        </a>
        <a
          aria-current={null}
          href="/user"
          onClick={[Function]}
        >
          <img
            alt="ProfileMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/e60b6c14d78fa03c5a1ae02b95f9cbca/user.png"
          />
        </a>
        <img
          alt="menu"
          className="nav__menu"
          onClick={[Function]}
          src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ea9ccf0f7a5fe6c8d066bb89f204f504/open-menu.png"
        />
      </div>
    </div>
    <div
      className="header__desktop"
    >
      <img
        alt="logo"
        className="logo"
        src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
      />
      <p
        className="appName"
      >
        bikUP
      </p>
      <ul
        className="header__navigation"
      >
        <a
          aria-current={null}
          className="navigation__item"
          href="/bikes"
          onClick={[Function]}
        >
          My Bikes
        </a>
        <a
          aria-current={null}
          className="navigation__item"
          href="/search"
          onClick={[Function]}
        >
          Explore
        </a>
        <a
          aria-current={null}
          className="navigation__item"
          href="/user"
          onClick={[Function]}
        >
          Profile
        </a>
      </ul>
      <div
        className="flex-spacer"
      />
      <button
        className="login__button login__button--header"
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </header>,
  <div
    className="general-container"
  >
    <div
      className="desktop"
    >
      <div
        className="standardaside"
      >
        <div
          className="standardaside__card-container"
        >
          <h3
            className="title"
          >
            Challenges
          </h3>
          <div
            className="standardaside__card"
          >
            <div>
              <div
                className="challenges__item"
              >
                <div
                  className="item__title"
                >
                  <img
                    alt="trophy"
                    className="title__img"
                    src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
                  />
                  - 
                  BIrlfriend
                   -
                </div>
                <p>
                  Travel 5000km with a bike
                </p>
              </div>
              <div
                className="separator"
              />
            </div>
            <div>
              <div
                className="challenges__item"
              >
                <div
                  className="item__title"
                >
                  <img
                    alt="trophy"
                    className="title__img"
                    src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
                  />
                  - 
                  CompoManiac
                   -
                </div>
                <p>
                  Change the same component 3 times
                </p>
              </div>
              <div
                className="separator"
              />
            </div>
          </div>
        </div>
        <div
          className="standardaside__card-container"
        >
          <h3
            className="title"
          >
            Give Away
          </h3>
          <div
            className="standardaside__card"
          >
            <p>
              something free
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="bike-detail"
    >
      <div
        className="bike-detail__upper mobile"
      >
        <a
          aria-current={null}
          href="/bikes"
          onClick={[Function]}
        >
          Back
        </a>
        <a
          aria-current={null}
          className="upper__edit"
          href="/bikes/undefined/edit"
          onClick={[Function]}
        >
          Edit
        </a>
      </div>
      <div
        className="bike-detail__head"
      >
        <div
          className="head__container"
        >
          <a
            aria-current={null}
            className="desktop"
            href="/bikes"
            onClick={[Function]}
          >
            Back
          </a>
          <h2
            className="head__bikename"
          >
            Your Bike
          </h2>
          <a
            aria-current={null}
            className="desktop"
            href="/bikes/undefined/edit"
            onClick={[Function]}
          >
            Edit
          </a>
        </div>
        <div
          className="separator"
        />
        <div
          className="bike-detail__km-hours"
        >
          <div
            className="km-hours km-hours__km"
          >
            <p
              className="km-hours__title"
            >
              Total KM
            </p>
            <div
              className="separator-small"
            />
            <p
              className="km__value"
            >
              NaN
            </p>
          </div>
          <div
            className="km-hours__hours km-hours"
          >
            <p
              className="km-hours__title"
            >
              Total Hours
            </p>
            <div
              className="separator-small"
            />
            <p
              className="km-hours__value"
            >
              NaN
            </p>
          </div>
          <button
            className="float__button"
            onClick={[Function]}
          >
            + Workout
          </button>
        </div>
      </div>
      <div
        className="bike-detail__components"
      />
      <div
        className="bike-detail__details"
      >
        <div
          className="details__content"
        >
          <img
            alt="bike type"
            className="details__type-img"
            src="https://image.flaticon.com/icons/svg/829/829906.svg"
          />
          <div
            className="details__info"
          >
            <p
              className="info__type"
            />
            <p>
              Unknown Brand
               
              -
               
              Unknown Model
            </p>
            <p>
              Driving Style:
               
            </p>
          </div>
        </div>
      </div>
      <div
        className="darken__back"
      />
      <div
        className="addworkout__container"
      >
        <div
          className="addworkout__top"
        >
          <div
            className="flex-spacer"
          />
          <h3>
            Add new workout
          </h3>
          <div
            className="flex-spacer"
          />
          <p
            className="top__close"
            onClick={[Function]}
          >
            X
          </p>
        </div>
        <div
          className="bikecard__separator"
        />
        <div
          className="addworkout__km-container addworkout__element"
        >
          <label
            className="addworkout__label"
            htmlFor="workoutKm"
          >
            Distance (KM)
          </label>
          <input
            className="addworkout__km-input addworkout__input"
            defaultValue="0"
            name="addworkout__km"
            required={true}
            type="number"
          />
        </div>
        <div
          className="addworkout__time-container addworkout__element-double"
        >
          <div
            className="time__hours"
          >
            <label
              className="addworkout__label"
              htmlFor="addworkout__hours"
            >
              Hours
            </label>
            <input
              className="addworkout__hours-input addworkout__input"
              defaultValue="0"
              name="addworkout__hours"
              required={true}
              type="number"
            />
          </div>
          <div
            className="time__minutes"
          >
            <label
              className="addworkout__label"
              htmlFor="addworkout__minutes"
            >
              Minutes
            </label>
            <input
              className="addworkout__minutes-input addworkout__input"
              defaultValue="0"
              name="addworkout__minutes"
              required={true}
              type="number"
            />
          </div>
        </div>
        <div
          className="addworkout__bottom"
        >
          <button
            className="addworkout__button"
            onClick={[Function]}
          >
            <div
              className="loading hidden"
            >
              <img
                alt="loading..."
                className="loading__img"
                src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f5753c458a8b552f891bb81/af512a8cb3c1285000d1191fdaaa670c/Spinner-1s-200px_(1).gif"
              />
              <p>
                Loading
              </p>
            </div>
            <div
              Account={true}
              Create={true}
              className="no-loading"
            >
              Add new Workout
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
